### PR TITLE
TRITON-2254 Change joyMattermostNotification to joySlackNotifications

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

**Review by papertigers — DISMISSED** [review]

**Comment by papertigers** [comment]
The failure here is due to an older rust version being used by jenkins. See issue #4 for the longer version.

**Comment by papertigers** [comment]
My only issue with expanding it back out to how it used to be is that anyone using it sees clippy warnings. 
I would much rather get this swapped over to a new image or eventually start relying on rustup.

If we are going to merge this the way you have it we should open a tracking issue to swap it back to `matches!()`

**Comment by bahamat** [comment]
Fair enough. Created #7 for that.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [package]
 name = "illumos-priv"
 version = "0.2.0"
+rust = "1.4.0"
```
Comment (papertigers) [inline Cargo.toml]: Does this actually work for you if you do `cargo check`, locally it told me ignoring unknown key package.rust

Comment (bahamat) [reply Cargo.toml]: Yeah, I'll yank it.

**Review by papertigers — APPROVED** [review]

**Pull request merged**